purchase_delivery_split_date/models/purchase.py

```diff
@@ -1,138 +1,100 @@
 # Copyright 2014-2016 Numérigraphe SARL
 # Copyright 2017 ForgeFlow, S.L.
+# Copyright 2021 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from itertools import groupby
+from odoo import api, fields, models, SUPERUSER_ID
 
-from odoo import api, fields, models
+
+def date(picking, datetime):
+    # TODO: extract the tz field on the warehouse from the module
+    # sale_cutoff_time_delivery in OCA/sale-workflow to make a generic module
+    # on which this module can depend on. At the moment, we take the tz of the
+    # SUPERUSER. This is safer than the tz of the user (purchaser)
```

Comment:
How is it safer ? Checking on the project those changes are intended for, I see the superuser has no tz set !

Reply:
I find it safer because it is not anymore based on a user that can freely change his timezone in his preferences.

Reply:
I'm adding a fallback on context tz or user tz if not defined on admin

Reply:
There is already kind of one in `context_today` https://github.com/odoo/odoo/blob/aff7694c56185212c17ceaf358ad99b3d14dee8b/odoo/fields.py#L1716